frontend/src/tests/instructor-permission-test.js

```diff
@@ -53,54 +206,238 @@ export function instructorsPermissionTests(api) {
         });
     });
 
-    it.todo("can't update instructors except for self (i.e. active user)");
+    it("can't update instructors except for self (i.e. active user)", async () => {
+        // we use the default user's admin privilege here to fetch an unrelated instructor
+        let respFetchInst = await apiGET(`/instructor/instructors`);
+        expect(respFetchInst).toHaveStatus("success");
+
+        let respFetchDefaultUser = await apiGET(`/instructor/active_user`);
+        expect(respFetchDefaultUser).toHaveStatus("success");
+        const defaultUser = respFetchDefaultUser.payload;
+
+        const defaultUserWithUpdatedInformation = { ...defaultUser };
+        defaultUserWithUpdatedInformation.email =
+            "instructor_permission_instructor_update_test@test.com";
+
+        // instructor can not modify other instructors
+        // in this example, the instructor only user cant update default user's information
+        await switchToInstructorOnlyUser();
+        let respInvalidRouteWithSession = await apiPOST(
+            `/instructor/instructors`,
+            defaultUserWithUpdatedInformation
+        );
+        expect(respInvalidRouteWithSession).toHaveStatus("error");
+        checkPropTypes(errorPropTypes, respInvalidRouteWithSession);
+
+        await restoreDefaultUser();
+
+        // instructors can modify themselves
+        let respModDefaultUser = await apiPOST(
+            `/instructor/instructors`,
+            defaultUserWithUpdatedInformation
+        );
+        expect(respModDefaultUser).toHaveStatus("success");
+        expect(respModDefaultUser.payload.email).toEqual(
+            "instructor_permission_instructor_update_test@test.com"
+        );
+    });
 
     it("fetch sessions", async () => {
-        let resp = await apiGET("/instructor/instructors");
+        await switchToInstructorOnlyUser();
+        let resp = await apiGET("/instructor/sessions");
+        expect(resp).toHaveStatus("success");
```

Comment:
What about the payload?

Reply:
I would argue checking the content of the response should be checked inside the instructor-tests.js, while intructor-permission-tests.js only checks the accessibility of different routes given a certain set of permissions.